src/platform/mbed/net/src/net_if.cpp

```diff
@@ -140,4 +143,103 @@ char * mbed_inet_ntop(sa_family_t family, const void * src, char * dst, size_t s
 int mbed_inet_pton(sa_family_t family, const char * src, void * dst)
 {
     return 0;
+}
+
+int mbed_ioctl(int fd, unsigned long request, void * param)
+{
+    int ret = 0;
+    switch (request)
+    {
+    case SIOCGIFFLAGS: {
+        if (param == NULL)
+        {
+            errno = EFAULT;
+            ret   = -1;
+            break;
+        }
+        struct ifreq * intfData = (struct ifreq *) param;
```

Comment:
I think that would be great to divide the ioctl into subfunction to avoid getting a 2000 line long functions as more support is added. `ioctl` should just served to verify input and dispatch the `request` to the right function.

Reply:
Sure, I am going to refactor this and use the socket to get network flags. Then I will divide it